packages/ref-metrics-ilamb/src/cmip_ref_metrics_ilamb/standard.py

```diff
@@ -143,43 +208,49 @@ def run(self, definition: MetricExecutionDefinition) -> MetricResult:
             csv_file = definition.to_output_path(f"{source_name}.csv")
             ref_file = definition.to_output_path("Reference.nc")
             com_file = definition.to_output_path(f"{source_name}.nc")
+            log_file = definition.to_output_path(f"{source_name}.log")
+            log_id = logger.add(log_file, backtrace=True, diagnose=True)
 
-            # Run the analysis on the data/model pair
-            com = xr.open_mfdataset(sorted(grp["path"].to_list()))
             try:
-                df_scalars, ds_ref, ds_com[source_name] = run_analyses(reference_dataset, com, analyses)
+                # Load data and run comparison
+                ref = _load_reference_data(variable, self.registry, setup["sources"], setup["relationships"])
+                com = _load_comparison_data(variable, grp)
+                dfs, ds_ref, ds_com[source_name] = run.run_analyses(ref, com, analyses)
+                dfs["source"] = dfs["source"].str.replace("Comparison", source_name)
+
+                # Write out artifacts
+                dfs.to_csv(csv_file, index=False)
+                if not ref_file.is_file():  # pragma: no cover
+                    ds_ref.to_netcdf(ref_file)
+                ds_com[source_name].to_netcdf(com_file)
+                df_all.append(dfs)
             except Exception:  # pragma: no cover
-                raise AnalysisFailure(self.name, source_name)
-            df_scalars["source"] = df_scalars["source"].str.replace("Comparison", source_name)
-            dfs += [df_scalars]
+                logger.exception(f"ILAMB analysis {self.slug} failed for {source_name}.")
```

Comment:
If no sources execute successful, should the execution should be deemed unsuccessful? If so the `MetricResult` returned from this func should `have the `successful` prop = False

Reply:
Ah yes, this is true. Will fix.